### PR TITLE
Cbonif/add-non-model-auth-section

### PR DESCRIPTION
**Pull request by chrisbonifacio**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by arundna — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/data/customize-authz/index.mdx [hunk 1]
```diff
@@ -144,6 +144,44 @@ const schema = a.schema({
...
+
+**Non-model** types are any types added to the schema without using `a.model()`. These consist of modifiers such as `a.customType()`, `a.enum()`,`a.query()`, `a.mutation()`, or `a.subscription()`.
+
+Dynamic authorization rules such as `allow.owner()`, `allow.ownerDefinedIn()`, `allow.groupDefinedIn()` are not supported for **non-model** types.
+
+**Example**
+
+```ts
+// Custom Query
+listCustomType: a
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/customize-authz/index.mdx:157]: does the syntax highlighting render okay? previous examples wrap with `a.schema()`

Comment (chrisbonifacio) [reply src/pages/[platform]/build-a-backend/data/customize-authz/index.mdx:157]: Yeah, looks okay on render:

CleanShot 2024-07-12 at 12 52 40@2x

Comment (josefaidt) [reply src/pages/[platform]/build-a-backend/data/customize-authz/index.mdx:157]: can we wrap it with `a.schema()` for consistency with the other examples?

Comment (chrisbonifacio) [reply src/pages/[platform]/build-a-backend/data/customize-authz/index.mdx]: Sure! pushed new commit

**Review by reesscot — APPROVED** [review]

**Pull request merged**